core/src/main/scala/org/apache/spark/rdd/NewHadoopRDD.scala

```diff
@@ -109,18 +109,22 @@ class NewHadoopRDD[K, V](
       logInfo("Input split: " + split.serializableHadoopSplit)
       val conf = confBroadcast.value.value
 
-      val inputMetrics = new InputMetrics(DataReadMethod.Hadoop)
+      val readMethod = DataReadMethod.Hadoop
```

Comment:
Minor, but I find it slightly easier to follow as a match:

```
      val hadoopReadMethod = DataReadMethod.Hadoop
      val newMetrics = context.taskMetrics.inputMetrics match {
        case Some(InputMetrics(hadoopReadMethod)) =>
          context.taskMetrics.inputMetrics
        case _ =>
          // Note that this may clobber some other input metric (see SPARK-XXX)
          hadoopReadMethod
      }
      context.taskMetrics.inputMetrics = Some(newMetrics)
```